### PR TITLE
Fixed typos in code comments

### PR DESCRIPTION
**Pull request by AndrewDryga**
There was number of typos in docs and comments, this PR fixes some of them.

### CODE REVIEW

File: src/rabbit_msg_store.erl [hunk 1]
```diff
@@ -244,11 +244,11 @@
 %%
 %% When we discover that a file is now empty, we delete it. When we
 %% discover that it can be combined with the useful data in either its
-%% left or right neighbour, and overall, across all the files, we have
+%% left or right neighbor, and overall, across all the files, we have
```
Comment (michaelklishin) [inline src/rabbit_msg_store.erl]: This is not a typo.

File: src/rabbit_msg_store.erl [hunk 2]
```diff
@@ -244,11 +244,11 @@
...
 %% ((the amount of garbage) / (the sum of all file sizes)) >
 %% ?GARBAGE_FRACTION, we start a garbage collection run concurrently,
 %% which will compact the two files together. This keeps disk
-%% utilisation high and aids performance. We deliberately do this
+%% utilization high and aids performance. We deliberately do this
```
Comment (michaelklishin) [inline src/rabbit_msg_store.erl]: This is not a typo either.

File: src/rabbit_msg_store.erl [hunk 3]
```diff
@@ -326,7 +326,7 @@
 %% is written several times we only store it once, and only remove it
 %% from the store when it has been removed the same number of times.
 %%
-%% The reference counts do not persist. Therefore the initialisation
+%% The reference counts do not persist. Therefore the initialization
```
Comment (michaelklishin) [inline src/rabbit_msg_store.erl]: Same as above.

File: src/rabbit_queue_index.erl [hunk 4]
```diff
@@ -50,7 +50,7 @@
 %%
 %% In order to be able to clean up ack'd messages, we write to segment
 %% files. These files have a fixed number of entries: ?SEGMENT_ENTRY_COUNT
-%% publishes, delivers and acknowledgements. They are numbered, and so
+%% publishes, delivers and acknowledgments. They are numbered, and so
```
Comment (michaelklishin) [inline src/rabbit_queue_index.erl]: Same as above.

**Comment by michaelklishin** [comment]
@AndrewDryga thank you but I cannot accept this. I'm not sure what spell checker you used but it assumes that British spelling of certain words is incorrect. Not to be a language purist but RabbitMQ has its roots in the UK and it should be respected by the core team we have today.

Some of the changes are indeed genuine typos. If you are still interested in seeing this merged, please submit a new PR with remaining corrections against the `stable` branch.

**Comment by AndrewDryga** [comment]
Actually I was using list of common typos from Wikipedia to grep over source code. After that I used to fix typos manually.

I will retry carefully.

**Pull request closed without merging**